template/add_funcs.go

```diff
@@ -49,9 +49,12 @@ func (t *Template) addFuncs() {
 	add(Git, t.addGitFuncs)
 }
 
-// Apply all regular expressions replacements to the supplied string
-func (t *Template) substitute(content string) string {
-	return utils.Substitute(content, t.substitutes...)
+// Apply regular expression that match the replacerTimingFilter
+//
+//  substitute(content string, SubstituteTimingsEnum.OneOfTheValues)
+// - Note: The best way to call this function is to use the utils.SubstituteTimingsEnum
+func (t *Template) substitute(content string, replacerTimingFilter utils.SubstituteTiming) string {
```

Comment:
Just a side note, it is pretty useless to document non public functions as they do not appear by default in the generated documentation.

Reply:
I documented it mostly for the next dev who's going to be in the code and is going to look at how to call this. In the end, I removed most of it since, with your recommandations with varargs, it became a lot less useful.